File: Phase01-Search/src/test/java/utils/DocumentProcessorTest.java
package utils;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashSet;
import java.util.Set;

class DocumentProcessorTest {
    @Test
    void getNormalizedWordsTest() {
        String firstTest  = "HeLLO WorlD!";
        String secondTest = "Hi I a.m A HumAn";
        String thirdTest = "the ExPe-cted tesTiNg is , the, expect, test, is";
        String[] firstTestExpected = {"hello", "world"};
        String[] secondTestExpected = {"hi", "i", "am", "a", "human"};
        String[] thirdTestExpected = {"the", "exp", "cted", "test", "is", "the", "expect", "test", "is"};
        Assertions.assertAll(() -> {
            Assertions.assertArrayEquals(firstTestExpected, new DocumentProcessor(firstTest).getNormalizedWords());
            Assertions.assertArrayEquals(secondTestExpected, new DocumentProcessor(secondTest).getNormalizedWords());
            Assertions.assertArrayEquals(thirdTestExpected, new DocumentProcessor(thirdTest).getNormalizedWords());
        });
    }

    @Test
    void DocumentProcessorConstructorTest() {
        Set<String> firstTest = new HashSet<>();
        firstTest.add("t&h*e");
        firstTest.add("aRt");
        firstTest.add("O@f");
        firstTest.add("Creations");
        firstTest.add("t*he");
        firstTest.add("jo&b");
        firstTest.add("foR#");
        firstTest.add("creators");
        String[] firstTestExpected = {"job", "art", "the", "of", "the", "creator", "creation", "for"};
        Assertions.assertArrayEquals(firstTestExpected, new DocumentProcessor(firstTest).getNormalizedWords());
    }
}

File: Phase08-EFCore/SearchLib/SearchLib.Test/Utils/DocumentProcessorTest.cs
using FluentAssertions;
using SearchLib.Utils;
using System.Collections.Generic;
using Xunit;

namespace SearchLib.Test.Utils
{
    public class DocumentProcessorTest
    {
        [Theory]
        [InlineData("HeLLO WorlD!", new string[] { "hello", "world" })]
        [InlineData("Hi I a.m A HumAn", new string[] { "hi", "i", "am", "a", "human" })]
        [InlineData("the ExPe-cted tesTiNg is , the, expect, test, is", new string[] { "the", "exp", "cted", "test", "is", "expect" })]
        void GetNormalizedWordsTest(string text, string[] expected)
        {
            string[] actual = new DocumentProcessor(text).GetNormalizedWords();
            actual.Should().BeEquivalentTo(expected);
        }

        [Fact]
        void DocumentProcessorConstructorTest()
        {
            string[] expected = { "job", "art", "the", "of", "creator", "creation", "for" };
            ISet<string> data = new HashSet<string>();
            data.Add("t&h*e");
            data.Add("aRt");
            data.Add("O@f");
            data.Add("Creations");
            data.Add("t*he");
            data.Add("jo&b");
            data.Add("foR#");
            data.Add("creators");
            string[] actual = new DocumentProcessor(data).GetNormalizedWords();
            actual.Should().BeEquivalentTo(expected);
        }
    }
}
